pycsob/utils.py

```diff
@@ -72,7 +77,11 @@ def dttm(format_='%Y%m%d%H%M%S'):
 def validate_response(response, key):
     response.raise_for_status()
 
-    data = response.json()
+    try:
+        data = response.json()
+    except JSONDecodeError:
```

Comment:
what other response errors?

Reply:
Added handling of other response errors, HTTP errors here and connection related errors in the HTTPAdapter.

Reply:
Better?